src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageSectorSelection/ImageSectorSelection.js

```diff
@@ -0,0 +1,130 @@
+import selectionStyles from './imageSectorSelection.scss';
+
+@observer
+export default class ImageSectorSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        src: string,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+
+    @observable imageLoaded = false;
+    @observable containerWidth: number;
+    @observable containerHeight: number;
+
+    naturalHorizontalToReal = (h: number) => h * this.imageResizedWidth / this.image.naturalWidth;
+    realHorizontalToNatural = (h: number) => h * this.image.naturalWidth / this.imageResizedWidth;
+    naturalVerticalToReal = (v: number) => v * this.imageResizedHeight / this.image.naturalHeight;
+    realVerticalToNatural = (v: number) => v * this.image.naturalHeight / this.imageResizedHeight;
+
+    naturalDataToReal(data: SelectionData): SelectionData {
+        return {
+            width: this.naturalHorizontalToReal(data.width),
+            height: this.naturalVerticalToReal(data.height),
+            left: this.naturalHorizontalToReal(data.left),
+            top: this.naturalVerticalToReal(data.top),
+        };
+    }
+
+    realDataToNatural(data: SelectionData): SelectionData {
+        return {
+            width: this.realHorizontalToNatural(data.width),
+            height: this.realVerticalToNatural(data.height),
+            left: this.realHorizontalToNatural(data.left),
+            top: this.realVerticalToNatural(data.top),
+        };
+    }
+
+    @action componentWillMount() {
+        this.image = new Image();
+        this.image.onload = action(() => this.imageLoaded = true);
+        this.image.onerror = () => log.error('Failed to preload image ' + this.props.src);
+        this.image.src = this.props.src;
+    }
+
+    @computed get imageResizedHeight(): number {
+        if (this.imageTouchesHoriziontalBorders()) {
+            return Math.min(this.image.naturalHeight, this.containerHeight);
+        } else {
+            return this.imageResizedWidth * this.image.naturalHeight / this.image.naturalWidth;
+        }
+    }
+
+    @computed get imageResizedWidth(): number {
+        if (this.imageTouchesHoriziontalBorders()) {
+            return this.imageResizedHeight * this.image.naturalWidth / this.image.naturalHeight;
+        } else {
+            return Math.min(this.image.naturalWidth, this.containerWidth);
+        }
+    }
+
+    imageTouchesHoriziontalBorders() {
+        const imageHeightToWidth = this.image.naturalHeight / this.image.naturalWidth;
+        const containerHeightToWidth = this.containerHeight / this.containerWidth;
+        return imageHeightToWidth > containerHeightToWidth;
+    }
+
+    handleRectangleSelectionChange = (data: SelectionData) => {
```

Comment:
At one point the data returned form the `RectangleSelection` has to be converted to be with respect to the natural dimensions of the loaded image. The `RectangleSelection` provides this data through a callback passed with the props. So in this component we have to define method which acts as a callback, takes the data and converts it.